subsys/bluetooth/controller/ll_sw/isoal.c

```diff
@@ -212,6 +224,110 @@ isoal_status_t isoal_sink_create(
+ *
+ * @param hdl[out]  Handle to source
+ * @return ISOAL_STATUS_OK if we could allocate; otherwise ISOAL_STATUS_ERR_SOURCE_ALLOC
+ */
+static isoal_status_t isoal_source_allocate(isoal_source_handle_t *hdl)
+{
+	isoal_source_handle_t i;
+
+	/* Very small linear search to find first free */
+	for (i = 0; i < ISOAL_SOURCES_MAX; i++) {
+		if (isoal_global.source_allocated[i] == ISOAL_ALLOC_STATE_FREE) {
+			isoal_global.source_allocated[i] = ISOAL_ALLOC_STATE_TAKEN;
+			*hdl = i;
+			return ISOAL_STATUS_OK;
+		}
+	}
+
+	return ISOAL_STATUS_ERR_SOURCE_ALLOC; /* All entries were taken */
+}
+
+/**
+ * @brief Mark a source as being free to allocate again
+ * @param hdl[in]  Handle to source
+ */
+static void isoal_source_deallocate(isoal_source_handle_t hdl)
+{
+	isoal_global.source_allocated[hdl] = ISOAL_ALLOC_STATE_FREE;
+}
+
+/**
+ * @brief Create a new source
+ *
+ * @param handle[in]            Connection handle
+ * @param role[in]              Peripheral, Central or Broadcast
+ * @param burst_number[in]      Burst Number
+ * @param flush_timeout[in]     Flush timeout
+ * @param max_octets[in]        Maximum PDU size (Max_PDU_C_To_P / Max_PDU_P_To_C)
+ * @param sdu_interval[in]      SDU interval
+ * @param iso_interval[in]      ISO interval
+ * @param stream_sync_delay[in] CIS sync delay
+ * @param group_sync_delay[in]  CIG sync delay
+ * @param pdu_alloc[in]         Callback of PDU allocator
+ * @param pdu_write[in]         Callback of PDU byte writer
+ * @param pdu_emit[in]          Callback of PDU emitter
+ * @param pdu_release[in]       Callback of PDU deallocator
+ * @param hdl[out]              Handle to new source
+ *
+ * @return ISOAL_STATUS_OK if we could create a new sink; otherwise ISOAL_STATUS_ERR_SOURCE_ALLOC
+ */
+isoal_status_t isoal_source_create(
+	uint16_t                    handle,
+	uint8_t                     role,
+	uint8_t                     burst_number,
+	uint8_t                     flush_timeout,
+	uint8_t                     max_octets,
+	uint32_t                    sdu_interval,
+	uint16_t                    iso_interval,
+	uint32_t                    stream_sync_delay,
+	uint32_t                    group_sync_delay,
+	isoal_source_pdu_alloc_cb   pdu_alloc,
+	isoal_source_pdu_write_cb   pdu_write,
+	isoal_source_pdu_emit_cb    pdu_emit,
+	isoal_source_pdu_release_cb pdu_release,
+	isoal_source_handle_t       *hdl)
+{
+	isoal_status_t err;
+
+	/* Allocate a new source */
+	err = isoal_source_allocate(hdl);
+	if (err) {
+		return err;
+	}
+
+	struct isoal_source_session *session = &isoal_global.source_state[*hdl].session;
+
+	session->handle = handle;
+
+	/* Todo: Next section computing various constants, should potentially be a
+	 * function in itself as a number of the dependencies could be changed while
+	 * a connection is active.
+	 */
+
+	/* Note: sdu_interval unit is uS, iso_interval is a multiple of 1.25mS */
+	session->pdus_per_sdu = burst_number * (sdu_interval / (iso_interval * 1250));
```

Comment:
Use `PERIODIC_INT_UNIT_US` or `CONN_INT_UNIT_US`
```suggestion
	session->pdus_per_sdu = burst_number * (sdu_interval / (iso_interval * PERIODIC_INT_UNIT_US));
```

Reply:
Done.

Reply:
@cvinayak Would be great if I could have a review of the update.
Thanks.